packages/jest-cli/src/SearchSource.js

```diff
@@ -183,6 +185,20 @@ class SearchSource {
     };
   }
 
+  findRelatedTestsFromPattern(
+    patternInfo: PatternInfo,
+    cwd: Path = process.cwd(),
+  ): SearchResult {
+    if (patternInfo.input.length) {
+      const sources = patternInfo.input.split(' ');
+      const resolvedPaths = sources.map(p => path.resolve(cwd, p));
+      return this.findRelatedTests(
+        new Set(Array.prototype.concat.apply([], resolvedPaths)),
```

Comment:
In this specific example, because you are dealing with a one-dimensional array, simply doing `new Set(resolvedPaths)` should be enough. The intent for this was to convert a nested array into a set of unique values.